pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java

```diff
@@ -0,0 +1,408 @@
+            + "FROM  somatic_documents sD "
+            + "LEFT JOIN ddp_instance as ddp on (ddp.ddp_instance_id = sD.ddp_instance_id) "
+            + "LEFT JOIN ddp_participant p on (p.participant_id = sD.participant_id) ";
+
+    private static final String SQL_SELECT_DOCUMENTS_BY_REALM = SQL_SELECT_DOCUMENTS_BASE
+            + "WHERE ddp.instance_name = ?";
+
+    private static final String SQL_SELECT_DOCUMENTS_BY_REALM_AND_PTPT = SQL_SELECT_DOCUMENTS_BASE
+            + "WHERE ddp.instance_name = ? AND ddp_participant_id = ?";
+
+    private static final String SQL_SELECT_DOCUMENT_BY_ID = SQL_SELECT_DOCUMENTS_BASE
+            + "WHERE sD.id = ?";
+
+    private static final String SQL_SELECT_DOCUMENT_BY_BUCKET_AND_PATH = SQL_SELECT_DOCUMENTS_BASE
+            + "WHERE sD.bucket = ? AND sD.blob_path = ?";
+
+    private static final String SQL_UPDATE_VIRUS_STATUS_SUCCESS = "UPDATE somatic_documents SET is_virus_free = ?, "
+            + "bucket = ?, blob_path = ? WHERE bucket = ? AND blob_path = ?";
+
+    private static final String SQL_UPDATE_VIRUS_STATUS_FAILED = "UPDATE somatic_documents SET is_virus_free = ?, "
+            + "deleted_at = ? WHERE bucket = ? AND blob_path = ?";
+
+    private static final String SQL_DELETE_DOCUMENT_BY_DOCUMENT_ID = "UPDATE somatic_documents SET deleted_by_user_id = ?, "
+            + "deleted_at = ? WHERE id = ?";
+
+    private static final String SQL_INSERT_SOMATIC_DOCUMENT = "INSERT INTO somatic_documents SET ddp_instance_id = "
+            + "(SELECT ddp_instance_id FROM ddp_instance WHERE instance_name = ?), file_name = ?, "
+            + "mime_type = ?, bucket = ?, blob_path = ?, created_by_user_id = ?, created_at= ?, "
+            + "participant_id = (SELECT p.participant_id from ddp_participant p "
+            + "LEFT JOIN ddp_instance as ddp ON (ddp.ddp_instance_id = p.ddp_instance_id) "
+            + "WHERE p.ddp_participant_id = ? AND ddp.instance_name= ?)";
+
+    /**
+     * Builds A Somatic Document for storage
+     *
+     * @param somaticDocumentId               document ID
+     * @param ddpInstanceId    Instance ID
+     * @param ddpParticipantId DDP Participant ID
+     * @param participantId    Participant ID
+     * @param fileName         Name of the file
+     * @param mimeType         Mime type of the file
+     * @param bucket           The bucket in GCS
+     * @param blobPath         Path of the file in GCS
+     * @param createdByUserId  User who uploaded the file
+     * @param createdAt        Epoch of when the file was uploaded
+     * @param deletedByUserId  User who deleted the file
+     * @param deletedAt        Epoch of when the file was deleted
+     * @param isVirusFree       True when the file has been through file scanning and moved to study bucket.
+     */
+    public SomaticResultUpload(Long somaticDocumentId, long ddpInstanceId, String ddpParticipantId, long participantId, String fileName,
```

Comment:
Looks like you skipped the DAO, DTO pattern?

Reply:
I did.  I wanted to keep this with the least amount of indirection needed - since the client gets all of the information at points it didn't make sense to me to translate between.  Plenty in DSM already skips the DAO, DTO pattern.  Happy to implement if you'd like to see it.

Reply:
Mostly curious because as you mentioned there are two patterns, and I've just been dealing with code that uses *both* patterns for the same table simultaneously. I don't feel particularly strongly about the pattern, just trying to avoid the situation where both are used for the same table.
As I look at this I don't see a strong need for a DTO, but this long call signature feels like a builder might be in future revisions.

Reply:
I boiled down to the same opinion.